fitnessllm_dataplatform/infrastructure/RedisConnect.py

```
@@ -0,0 +1,105 @@
+        try:
+            self.interface.set(name=key, value=json.dumps(value))
+            if ttl:
+                self.interface.setex(name=key, value=json.dumps(value), time=ttl)
+                logger.debug(f"Wrote key with ttl {ttl}")
+            logger.debug(f"Set {key} to redis")
+
+        except redis.RedisError as exc:
+            logger.error(f"Failed to set key '{key}': {exc}")
+        finally:
+            self.close_connection()
+            return
```

Ruff (0.8.2)

`return` inside `finally` blocks cause exceptions to be silenced

(B012)